### DISCUSSION TITLE
Multiple modals on a page -- components not properly behind background

### DISCUSSION THREAD

**Discussion by denishaskin** [excid3/tailwindcss-stimulus-components#210] [Q&A] [answered]
There seem to be at least one or two other issues regarding a similar problem, but they seem to be ignored.

Using the [modal component](https://github.com/excid3/tailwindcss-stimulus-components/blob/master/docs/modal.md), if you have multiple of them on a page, opening a modal will not correctly hide components of later modals from showing up through the background.

A codepen that demonstrates this is here: https://codepen.io/denishaskin/pen/LYqZyja

![image](https://github.com/excid3/tailwindcss-stimulus-components/assets/976900/6fdbf168-180b-4bc9-a036-a1d56fd83556)

![image](https://github.com/excid3/tailwindcss-stimulus-components/assets/976900/3117e4ea-1bc1-43c4-9605-bfad3a183f1b)

The simplest example is just have 2 separate modal components, copying the HTML exactly from the project's examples.  Click on the first button, the modal will show but the second button also shows.

It is exacerbated if you have (as we do), other components that we have to have inside a modal component (for interaction with the controller).  Even if we didn't need that, the fact that the later modal's button isn't hidden is a non-starter.  Is there a way to work around this?

**Comment by excid3** [comment, marked as answer]
You've got a z-10 on that element, so you'll want to adjust so the modal backdrop is a higher z-index.

**Reply by denishaskin** [reply]
Well, that was easy. Thanks.

Note, I took that verbatim from the sample code here: https://github.com/excid3/tailwindcss-stimulus-components/blob/master/docs/modal.md

I just assumed the z-10 on the element was required for the component to work correctly.  You might to update the doc and remove it if possible?
